testPeerRecoveryTrimsLocalTranslog fails reporting "transport client is closed"

For instance, [this build](https://gradle-enterprise.elastic.co/s/g65qxmmhdg4nu/console-log?task=:server:integTest) failed like this:

```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=2136, name=Thread-24, state=RUNNABLE, group=TGRP-IndexRecoveryIT]
	at __randomizedtesting.SeedInfo.seed([7645CA1AB2A6B2BE:8BC3885856D2FE62]:0)
Caused by: java.lang.IllegalStateException: transport client is closed
	at __randomizedtesting.SeedInfo.seed([7645CA1AB2A6B2BE]:0)
	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:246)
	at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:57)
	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:394)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:396)
	at org.elasticsearch.client.FilterClient.doExecute(FilterClient.java:65)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:396)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:385)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:45)
	at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:52)
	at org.elasticsearch.indices.recovery.IndexRecoveryIT.lambda$testPeerRecoveryTrimsLocalTranslog$38(IndexRecoveryIT.java:1559)
	at java.lang.Thread.run(Thread.java:748)
```

I see a few similar failures over the last several weeks.

--- a/server/src/test/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java
+++ b/server/src/test/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java
@@ -1558,7 +1558,7 @@ public class IndexRecoveryIT extends ESIntegTestCase {
                             .setSource(Collections.singletonMap("f" + randomIntBetween(1, 10), randomNonNegativeLong()), XContentType.JSON)
                             .get();
                         assertThat(response.getResult(), isOneOf(CREATED, UPDATED));
-                    } catch (ElasticsearchException ignored) {
+                    } catch (IllegalStateException | ElasticsearchException ignored) {
                     }
                 }
             });

